org.eclipse.winery.frontends/app/tosca-management/src/app/wineryLicenseModule/wineryLicense.component.ts

```diff
@@ -29,66 +36,254 @@ import { Observable } from 'rxjs/Rx';
+                branch: ['']
+            });
+            this.secondFormGroup = this.formBuilder.group({
+                secondCtrl: ['', Validators.required]
+            });
+
+            // when license engine is used, get licenses from there
+            observables.push(this.service.getAllLicenses().pipe(
+                map(
+                    (licenses) => {
+                        this.options = licenses;
+                    })
+            ));
+        } else {
+            // else, get licenses from the local files
+            observables.push(
+                this.service.loadLocalLicenses()
                     .pipe(
-                        map(
-                            data => {
-                                this.licenseText = data;
-                                this.initialLicenseText = data;
-                            }),
                         catchError((e) => {
-                            this.handleMissingLicense();
-                            return Observable.of(null);
-                        })
-                    ),
-                this.service.loadLicenses()
-                    .pipe(
-                        catchError(e => {
                             this.handleError(e);
                             return Observable.of(null);
                         })
-                    )
-            ];
+                    ));
+            this.options = this.service.getLicenseNames();
+        }
+
+        forkJoin(observables).subscribe(() => {
+            this.loading = false;
+            this.selectedOptions = this.options;
+        });
+    }
+
+    onSubmitTemplateBased() {
+        const toscaElements = this.sharedData.path.split('/');
+        const toscaElementID = toscaElements[toscaElements.length - 1];
+        this.service.createSoftware(toscaElementID, this.firstFormGroup.value['link'], this.firstFormGroup.value['branch']);
+
+        this.loadingbar = true;
+        this.firstFormGroup.disable();
+
+        this.service.deleteId().subscribe((success) => {
+            if (success) {
+                this.postSoftware();
+            } else {
+                this.failed();
+            }
+        }, () => {
+            this.failed();
+        });
+    }
+
+    postSoftware() {
+        this.service.postSoftware().subscribe(
+            (success) => {
+                if (success) {
+                    this.service.poll().subscribe(undefined, undefined, () => {
+                        this.checkLicenseData();
+                    });
+                } else {
+                    this.failed();
+                }
+            }, () => {
+                this.failed();
+            });
+    }
 
-        forkJoin(observables).subscribe();
+    checkLicenseData() {
+        if (this.service.isFinished()) {
+            this.foundLicenses = 'Extracted Licenses From Source Code: ';
+            this.foundLicenses += this.service.getFoundLicenses();
```

Comment:
This line can be inlined in the above line.

Reply:
changed